worksheet_D/oxo.py

```diff
@@ -0,0 +1,102 @@
+class OxoBoard:
+    def __init__(self):
+        self.gameboard = [[0, 0, 0], [0, 0, 0], [0, 0, 0]] #Initialises the gameboard list
+
+    def get_square(self, x, y):
+        return self.gameboard[x][y] #Returns the value of a square on the gameboard
+
+    def set_square(self, x, y, mark):
+        if self.gameboard[x][y] == 0:  #Checks if a square is empty, if so it's filled with mark, if not False is returned
+            self.gameboard[x][y] = mark
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
+        for x in range(3):  #Returns a false boolean if any squares on the board are empty
+            for y in range(3):
+                if self.gameboard[x][y] == 0:
+                    return False
+        return True
```

Comment:
May be worth adding a comment here e.g. "If we reach this point without finding an empty square, the board must be full"